[Feature] Support RTMDet-Ins and improve RTMDet test config. (#9494)
open-mmlab/mmdetection: README.md

> 

Reviewer: Also update the section of `state of the art` section in major features.

Reviewer: The newly released [RTMDet](configs/rtmdet) also obtains new state-of-the-art results on real-time instance segmentation and rotated object detection tasks and the best parameter-accuracy trade-off on object detection.

## What's New

### Highlight

> We are excited to announce our latest work on real-time object recognition tasks, **RTMDet**, a family of fully convolutional single-stage detectors. RTMDet not only achieves the best parameter-accuracy trade-off on object detection from tiny to extra-large model sizes, but also becomes the state-of-the-art on instance segmentation and rotated object detection tasks. Detailed explanation of its methodologies is available in our [technical report](https://arxiv.org/abs/2212.07784). Pre-trained models can be found [here](configs/rtmdet).

Reviewer: ```suggestion
We are excited to announce our latest work on real-time object recognition tasks, **RTMDet**, a family of fully convolutional single-stage detectors. RTMDet not only achieves the best parameter-accuracy trade-off on object detection from tiny to extra-large model sizes but also obtains new state-of-the-art performance on instance segmentation and rotated object detection tasks. Details can be found in the [technical report](https://arxiv.org/abs/2212.07784). Pre-trained models are [here](configs/rtmdet).
```

[![PWC](https://img.shields.io/endpoint.svg?url=https://paperswithcode.com/badge/rtmdet-an-empirical-study-of-designing-real/real-time-instance-segmentation-on-mscoco)](https://paperswithcode.com/sota/real-time-instance-segmentation-on-mscoco?p=rtmdet-an-empirical-study-of-designing-real)
[![PWC](https://img.shields.io/endpoint.svg?url=https://paperswithcode.com/badge/rtmdet-an-empirical-study-of-designing-real/object-detection-in-aerial-images-on-dota-1)](https://paperswithcode.com/sota/object-detection-in-aerial-images-on-dota-1?p=rtmdet-an-empirical-study-of-designing-real)
[![PWC](https://img.shields.io/endpoint.svg?url=https://paperswithcode.com/badge/rtmdet-an-empirical-study-of-designing-real/object-detection-in-aerial-images-on-hrsc2016)](https://paperswithcode.com/sota/object-detection-in-aerial-images-on-hrsc2016?p=rtmdet-an-empirical-study-of-designing-real)

| Task                     | Dataset | AP                                   | FPS(TRT FP16 BS1 3090) |
| ------------------------ | ------- | ------------------------------------ | ---------------------- |
| Object Detection         | COCO    | 52.8                                 | 322                    |

> | Instance Segmentation    | COCO    | 44.6                                 | 188                    |

Reviewer 2: @RangiLyu

Unlike Mask R-CNN which outputs`n x 28 x 28` RoI masks if we skip post processing, RTMDet-Ins outputs `n x img_h x img_w` by default. I think transferring those large masks from GPU to CPU will take quite a bit of time. Is the time included in latency/FPS measure?

Author: The decoding and rescaling of masks are included, but the GPU-CPU transferring is not. We tested the latency with `trtexec` and took the GPU Compute Time as the result (like the conventions in the previous YOLO series).

Reviewer 2: Thanks for the reply. It seems we need some modifications of the output masks for application.

Anyway, the mAP and GPU compute time is impressive. The model looks promising.
